crates/neon-runtime/src/napi/date.rs

```diff
@@ -0,0 +1,27 @@
+use std::mem::MaybeUninit;
+use crate::napi::bindings as napi;
+use crate::raw::{Env, Local};
+
+/// Create a new date object
+///
+/// # Safety
+///
+/// `env` is a raw pointer. Please ensure it points to a napi_env that is valid for the current context.
+pub unsafe fn new_date(env: Env, value: f64) -> Local {
+    let mut local = MaybeUninit::zeroed();
+    let status = napi::create_date(env, value, local.as_mut_ptr());
+    assert_eq!(status, napi::Status::Ok);
+    local.assume_init()
+}
+
+/// Get the value of a date object
+///
+/// # Safety
+///
+/// `env` is a raw pointer. Please ensure it points to a napi_env that is valid for the current context.
```

Comment:
This one has an additional safety requirement that `Local` be a `napi_value` associated with the `Env`.